### PR TITLE
🔨 Heavy refactor

### PR DESCRIPTION
**Pull request by aendrew**
`lib/` moved to separate repo, [financial-times/s3o-middleware-utils][1]. This enables me to create a Koa 2 middleware with its own repo and `package.json` while allowing it to share common concerns with this repo. 

My automated tests all pass, but I haven't ever tried this in production. As such, it'd be appreciated if someone would be willing to do some ad-hoc integration testing.

Also, it's probably worth bumping the major version number as this will break anything relying on anything in `lib/`.

[1]: https://github.com/Financial-Times/s3o-middleware-utils

### CODE REVIEW

**Comment by aendrew** [comment]
I'm not sure why VersionEye is failing it, I think it has an old cached version of s3o-middleware-utils without a license.

File: package.json [hunk 1]
```diff
@@ -24,10 +24,8 @@
   "homepage": "https://github.com/Financial-Times/S3O-middleware#readme",
   "dependencies": {
     "body-parser": "^1.14.1",
-    "cookie": "^0.2.2",
     "debug": "^2.2.0",
-    "ft-poller": "^2.1.0",
-    "node-rsa": "^0.2.25"
+    "s3o-middleware-utils": "github:financial-times/s3o-middleware-utils"
```
Comment (JakeChampion) [inline package.json]: could this be published to npm under the @financial-times account please

Comment (aendrew) [inline package.json]: @JakeChampion Definitely, mind pinging me on Slack for the process whereby I'd do that? Thanks!

Comment (JakeChampion) [inline package.json]: @aendrew Has this been done?

Comment (aendrew) [inline package.json]: @JakeChampion Not yet, have been swamped with a bunch of other projects. Will try to get to that later today!

**Pull request merged**